### ISSUE TITLE
highspy: Passing user_callback_data to setCallback raises exception

### ISSUE THREAD

**Issue by few** [ERGO-Code/HiGHS#1904]
The following code raises an exception (at least on the `latest` branch):

```import highspy

h = highspy.Highs()

def callback(callback_type, message, data_out, data_in, user_callback_data):
    pass

user_callback_data = {}
h.setCallback(callback, user_callback_data)
```

```h.setCallback(callback, callback_data)
TypeError: setCallback(): incompatible function arguments. The following argument types are supported:
    1. (self: highspy._core._Highs, arg0: Callable[[int, str, HighsCallbackDataOut, HighsCallbackDataIn, capsule], None], arg1: capsule) -> highspy._core.HighsStatus

Invoked with: <highspy.highs.Highs object at 0x7f9b12ffa160>, <function callback at 0x7f9b132b7d90>, {}
```
Passing None as user_callback_data works.

**Comment by jajhall** [comment]
This is done so that users can pass data required in a callback. It works fine in C++ (and C, I think). I've got no idea how to make it work in Python

**Comment by few** [comment]
I found a workaround:
```
import highspy

h = highspy.Highs()
h.readModel("30n20b8.mps")
h.setOptionValue("time_limit", 90)

class CallbackManager:
    mip_solution = None

    def callback(self, callback_type, message, data_out, data_in, user_callback_data):
        self.mip_solution = data_out.mip_solution

callback_manager = CallbackManager()
h.setCallback(callback_manager.callback, None)
h.startCallback(highspy.cb.HighsCallbackType.kCallbackMipImprovingSolution)
h.run()
h.stopCallback(highspy.cb.HighsCallbackType.kCallbackMipImprovingSolution)

print("best solution so far:", callback_manager.mip_solution)
```

This way one can store arbitrary data in the CallbackManager object. Maybe user_callback_data is not needed in highspy?

**Comment by jajhall** [comment]
Closed by #1942
